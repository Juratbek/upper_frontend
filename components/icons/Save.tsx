import { IIconProps } from './types';

export const SaveIcon = ({ width = 24, height = 24, color = 'black' }: IIconProps) => (
  <svg width={width} height={height} viewBox='0 0 24 24' fill='none' role='icon'>
    <path
      d='M16.0166 3.66675H7.98327C6.20827 3.66675 5 4.73333 5 6.5V21.5L11.2249 17.7751C11.6583 17.5334 12.3583 17.5334 12.7833 17.7751L19 21.5V6.5C18.9917 4.73333 17.7916 3.66675 16.0166 3.66675Z'
      fill='white'
      fillOpacity='0.2'
      stroke={color}
      strokeWidth='1.6'
      strokeLinecap='round'
      strokeLinejoin='round'
    />
  </svg>
);
