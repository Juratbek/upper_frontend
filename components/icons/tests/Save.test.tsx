import { render, screen } from '@testing-library/react';

import { SaveIcon } from '../Save';

describe('Save icon', () => {
  it('renders the icon', () => {
    render(<SaveIcon />);
    const icon = screen.getByRole('icon');
    expect(icon).toMatchSnapshot();
  });
});
