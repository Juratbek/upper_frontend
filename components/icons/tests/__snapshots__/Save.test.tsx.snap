// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Save icon > renders the icon 1`] = `
<svg
  fill="none"
  height="24"
  role="icon"
  viewBox="0 0 24 24"
  width="24"
>
  <path
    d="M16.0166 3.66675H7.98327C6.20827 3.66675 5 4.73333 5 6.5V21.5L11.2249 17.7751C11.6583 17.5334 12.3583 17.5334 12.7833 17.7751L19 21.5V6.5C18.9917 4.73333 17.7916 3.66675 16.0166 3.66675Z"
    fill="white"
    fill-opacity="0.2"
    stroke="black"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.6"
  />
</svg>
`;
